File: src/main/java/com/dnd_fan/dnd_program/MainWindow.java
package com.dnd_fan.dnd_program;

import javafx.stage.Stage;

public class MainWindow {
    private static Stage sMainStage;

    public static void set(Stage mainStage) {
        sMainStage = mainStage;
    }

    public static Stage get() {
        return sMainStage;
    }
}

File: src/main/java/lt/dndfan/dndhelper/MainWindow.kt
package lt.dndfan.dndhelper

import javafx.application.Application
import javafx.stage.Stage
import lt.dndfan.dndhelper.util.gui.IWindowLoader
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.AnnotationConfigApplicationContext
import org.springframework.stereotype.Component
import javax.annotation.PostConstruct

fun main(args: Array<String>) {
    Application.launch(MainWindow().javaClass, *args)
}

@Component
class MainWindow : Application() {
    @Autowired
    private lateinit var windowLoader : IWindowLoader

    @Throws(Exception::class)
    override fun start(primaryStage: Stage) {
        val context = AnnotationConfigApplicationContext()
        context.register(AppConfig::class.java)
        context.refresh()
    }

    @PostConstruct
    private fun postConstructApplication() {
        windowLoader.getWindow("main_menu").render()
    }
}